### ISSUE TITLE
can you write a swift version

### ISSUE THREAD

**Issue by taylortran19** [louis49/androidtv-remote#29]
can you write a swift version.
I try to write a swift version but the socket don't work. tks you

**Comment by louis49** [comment]
Of course !  what is your method of payment?

**Comment by taylortran19** [comment]
give me your email. tks @louis49

**Comment by odyshewroman** [comment]
I already implement it in Swift without using protobuff or any third party library, just only native Swift, you can also message me

**Comment by Goule** [comment]
Hello @odyshewroman, 

How can I contact you? 
I don't see any link in your profile, thanks !

**Comment by odyshewroman** [comment]
> Hello @odyshewroman,
> 
> How can I contact you? I don't see any link in your profile, thanks !

odyshewroman@gmail.com

**Comment by hieuttmatech** [comment]
> I already implement it in Swift without using protobuff or any third party library, just only native Swift, you can also message me

and your code can send text and voice (google assistant) ?

**Comment by odyshewroman** [comment]
> > I already implement it in Swift without using protobuff or any third party library, just only native Swift, you can also message me
> 
> and your code can send text and voice (google assistant) ?

Nope, I use it for another reason, for sending command from iPhone to running necessary application on Android TV OS TV box. Then communicate with this application by own private protocol. But it's already details of your using this ability. In anyway I not interact with google assistant.

**Comment by mirroringcontact** [comment]
> Of course ! what is your method of payment?

Can you write me? i need swift version of this project.  screen.mirroring.contact@gmail.com

**Comment by mirroringcontact** [comment]
I have working swift iOS project.  https://t.me/vitrrr

**Comment by wh1teSp** [comment]
> Of course ! what is your method of payment?

Hi, I would be interested in purchasing, how can I contact you?

**Comment by astrokin** [comment]
@wh1teSp @Goule  @taylortran19  you can contact me if you have any question about implementation for iOS

**Comment by wh1teSp** [comment]
> @wh1teSp @Goule @taylortran19 you can contact me if you have any question about implementation for iOS

@astrokin how can I contact you ?

**Comment by astrokin** [comment]
all possible options you can find on profile page https://github.com/astrokin

**Comment by wh1teSp** [comment]
> all possible options you can find on profile page https://github.com/astrokin

@astrokin check on X, my account @t3ix

**Comment by astrokin** [comment]
@wh1teSp feel free to email me alex.strok@gmail.com

**Comment by wh1teSp** [comment]
Thanks @astrokin for your help

**Comment by odyshewroman** [comment]
ok guys, I have opened my first version for iOS - https://github.com/odyshewroman/AndroidTVRemoteControl

**Comment by astrokin** [comment]
feel free to find text input implementation without ADB for Android TV in [python](https://github.com/tronikos/androidtvremote2/pull/25) and [js](https://github.com/louis49/androidtv-remote/pull/46)

**Comment by starscodermoh** [comment]
@astrokin hey , i have an issue with this. it replace the text when on screen keyboard is open on tv can you please guid me how did you reverse engineer this or did your research so i can solve my issue , i have tried decompiling other applications that work correct but. no success

**Comment by astrokin** [comment]
@starscodermoh double check parameters you've initialised with imeObject
